aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -96,10 +181,13 @@ void LayerNormKernelImpl(
     Tensor* Y,
     Tensor* mean,
     Tensor* rstd) {
+  DCHECK_EQ(X.numel(), M * N);
+  DCHECK(!gamma.defined() || gamma.numel() == N);
+  DCHECK(!beta.defined() || beta.numel() == N);
   AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::BFloat16, X.scalar_type(),
       "LayerNormKernelImpl", [&]() {
     LayerNormKernelImplInternal<scalar_t>(
```

Comment:
and it indeed it looks like it's being run at low precision here

Reply:
Now the dispatch pattern is like:

* input is float, param is float: it will call `template <typename T> LayerNormKernelImplInternal` and `T` is float.

* input is bfloat16, param is bfloat16, it will call `template <>LayerNormKernelImplInternal<BFloat16>`, and it will go to `template <typename param_t>layer_norm_kernel_mixed_type` and `param_t` is BFloat16.

* input is bfloat16, param is float, it will call `template <>LayerNormKernelImplInternal<BFloat16>`, and it will go to `template <typename param_t>layer_norm_kernel_mixed_type` and `param_t` is float.

Sorry for the late response, working on some difficult tasks last few days.